### PR TITLE
feat(gatsby): Support absolute certificate paths for development

### PR DESCRIPTION
**Pull request by cranberyxl**
## Description
All certificate paths are currently relative to the project directory. This change makes it support `/` and `~/`

```gatsby develop --https --key-file ~/.certs/star.dev.key --cert-file ~/.certs/star.dev.crt``` 

Before: `/Users/cranberyxl/gatsby-project/Users/cranberyxl/.certs/star.dev.key`

After: `/Users/cranberyxl/.certs/star.dev.key`

### CODE REVIEW

File: packages/gatsby/src/utils/get-ssl-cert.js [hunk 1]
```diff
@@ -3,6 +3,14 @@ const report = require(`gatsby-cli/lib/reporter`)
 const fs = require(`fs`)
 const path = require(`path`)
 
+const resolvePath = (directory, filePath) => {
+  // Support absolute paths
+  if (filePath[0] === `/`) {
```
Comment (DSchau) [inline packages/gatsby/src/utils/get-ssl-cert.js]: How about [path.isAbsolute](https://nodejs.org/api/path.html#path_path_isabsolute_path)?

```suggestion
  if (path.isAbsolute(filePath[0])) {
```

Also why just one part of the path? I'm a little confused by that.

Comment (cranberyxl) [reply packages/gatsby/src/utils/get-ssl-cert.js]: `filePath` is a string, so I was looking at the first character. I'll adjust it to use the path method

**Comment by wardpeet** [comment]
Published in `gatsby@2.11.0`

**Comment by aaronjensen** [comment]
This is great, though afaict, self signed certs don't work anymore: https://github.com/gatsbyjs/gatsby/issues/14990

**Pull request merged**